src/waylandhelper.h

```diff
@@ -0,0 +1,38 @@
+#include <QtCore>
+
+#include <KWayland/Client/connection_thread.h>
+#include <KWayland/Client/registry.h>
+#include <KWayland/Client/surface.h>
+#include <KWayland/Client/xdgforeign.h>
+
+using namespace KWayland::Client;
+
+#ifndef WAYLANDHELPER_H
+#define WAYLANDHELPER_H
+
+class WaylandHelper : public QObject
+{
+    Q_OBJECT
+    Q_DISABLE_COPY(WaylandHelper)
+
+public:
+    explicit WaylandHelper(QWindow *parent);
+
+    QString exportedHandle();
+
+signals:
+    void doneXdgExport();
+
+private:
+    Surface *surface;
+    Registry registry;
+    ConnectionThread *connection;
+    XdgExporter *xdgExporter = nullptr;
+    XdgExported *xdgExportedTopLevel = nullptr;
+
+private slots:
+    void onXdgExporterAnnounced(uint name, uint version);
```

Comment:
Use `quint32`, the same typedef that in the signal:

```suggestion
    void onXdgExporterAnnounced(quint32 name, quint32 version);
```